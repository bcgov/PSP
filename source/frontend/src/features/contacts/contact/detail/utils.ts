import { ContactInfoField } from 'features/contacts/interfaces';
import { Dictionary } from 'interfaces/Dictionary';
import { IContactAddress, IContactMethod } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export function getContactInfo(
  validTypes: Dictionary<string>,
  contactMethods?: IContactMethod[],
): ContactInfoField[] {
  if (contactMethods === undefined) {
    return [];
  }
  // Get only the valid types
  let filteredFields = contactMethods.reduce(
    (accumulator: ContactInfoField[], method: IContactMethod) => {
      if (Object.keys(validTypes).includes(method.contactMethodType.id)) {
        accumulator.push({
          info: method.value,
          label: validTypes[method.contactMethodType.id],
        });
      }
      return accumulator;
    },
    [],
  );

  // Sort according to the dictionary order
  return filteredFields.sort((a, b) => {
    return Object.values(validTypes).indexOf(a.label) - Object.values(validTypes).indexOf(b.label);
  });
}

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
