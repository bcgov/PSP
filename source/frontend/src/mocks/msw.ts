import {
  getMockAddresses,
  getMockLandChars,
  getMockLegalDescriptions,
  getMockValues,
} from 'mocks/bcAssessmentMock';
// src/mocks.js
// 1. Import the library.
import { rest, setupWorker } from 'msw';

import { getMockDescription, getMockSales } from './bcAssessmentMock';

// 2. Describe network behavior with request handlers.
export const worker = setupWorker(
  rest.get('https://delivery.apps.gov.bc.ca/ext/sgw/geo.bca', (req, res, ctx) => {
    if (req.url.search.includes('BCA_FOLIO_ADDRESSES_SV')) {
      return res(ctx.delay(500), ctx.status(200), ctx.json(getMockAddresses()));
    } else if (req.url.search.includes('BCA_FOLIO_LEGAL_DESCRIPTS_SV')) {
      return res(ctx.delay(500), ctx.status(200), ctx.json(getMockLegalDescriptions()));
    } else if (req.url.search.includes('BCA_FOLIO_GNRL_PROP_VALUES_SV')) {
      return res(ctx.delay(500), ctx.status(200), ctx.json(getMockValues()));
    } else if (req.url.search.includes('BCA_FOLIO_SALES_SV')) {
      return res(ctx.delay(500), ctx.status(200), ctx.json(getMockSales()));
    } else if (req.url.search.includes('BCA_FOLIO_DESCRIPTIONS_SV')) {
      return res(ctx.delay(500), ctx.status(200), ctx.json(getMockDescription()));
    } else if (req.url.search.includes('BCA_FOLIO_LAND_CHARS_SV')) {
      return res(ctx.delay(500), ctx.status(200), ctx.json(getMockLandChars()));
    }
  }),
);
