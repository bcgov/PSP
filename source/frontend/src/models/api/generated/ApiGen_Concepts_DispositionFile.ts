/**
 * File autogenerated by TsGenerator.
 * Do not manually modify, changes made to this file will be lost when this file is regenerated.
 */
import { ApiGen_Base_CodeType } from './ApiGen_Base_CodeType';
import { ApiGen_Concepts_DispositionFileAppraisal } from './ApiGen_Concepts_DispositionFileAppraisal';
import { ApiGen_Concepts_DispositionFileOffer } from './ApiGen_Concepts_DispositionFileOffer';
import { ApiGen_Concepts_DispositionFileProperty } from './ApiGen_Concepts_DispositionFileProperty';
import { ApiGen_Concepts_DispositionFileSale } from './ApiGen_Concepts_DispositionFileSale';
import { ApiGen_Concepts_DispositionFileTeam } from './ApiGen_Concepts_DispositionFileTeam';
import { ApiGen_Concepts_FileWithChecklist } from './ApiGen_Concepts_FileWithChecklist';

// LINK: @backend/apimodels/Models/Concepts/DispositionFile/DispositionFileModel.cs
export interface ApiGen_Concepts_DispositionFile extends ApiGen_Concepts_FileWithChecklist {
  fileReference: string | null;
  initiatingDocumentDate: string | null;
  assignedDate: string | null;
  completionDate: string | null;
  dispositionTypeCode: ApiGen_Base_CodeType<string> | null;
  dispositionStatusTypeCode: ApiGen_Base_CodeType<string> | null;
  initiatingBranchTypeCode: ApiGen_Base_CodeType<string> | null;
  physicalFileStatusTypeCode: ApiGen_Base_CodeType<string> | null;
  fundingTypeCode: ApiGen_Base_CodeType<string> | null;
  initiatingDocumentTypeCode: ApiGen_Base_CodeType<string> | null;
  dispositionTypeOther: string | null;
  initiatingDocumentTypeOther: string | null;
  regionCode: ApiGen_Base_CodeType<number> | null;
  fileProperties: ApiGen_Concepts_DispositionFileProperty[] | null;
  dispositionTeam: ApiGen_Concepts_DispositionFileTeam[] | null;
  dispositionOffers: ApiGen_Concepts_DispositionFileOffer[] | null;
  dispositionSale: ApiGen_Concepts_DispositionFileSale | null;
  dispositionAppraisal: ApiGen_Concepts_DispositionFileAppraisal | null;
}
