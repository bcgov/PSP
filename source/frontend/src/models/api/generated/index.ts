/**
 * This is a TypeGen auto-generated file.
 * Any changes made to this file can be lost when this file is regenerated.
 */

export * from './ApiGen_AccessRequest';
export * from './ApiGen_AcquisitionFile';
export * from './ApiGen_AcquisitionFileChecklistItem';
export * from './ApiGen_AcquisitionFileChecklistItemType';
export * from './ApiGen_AcquisitionFileOwner';
export * from './ApiGen_AcquisitionFileProperty';
export * from './ApiGen_AcquisitionFileTeam';
export * from './ApiGen_Address';
export * from './ApiGen_Agreement';
export * from './ApiGen_BaseAudit';
export * from './ApiGen_BaseConcurrent';
export * from './ApiGen_Claim';
export * from './ApiGen_CodeType';
export * from './ApiGen_CompensationFinancial';
export * from './ApiGen_CompensationRequisition';
export * from './ApiGen_ConsultationLease';
export * from './ApiGen_Contact';
export * from './ApiGen_ContactMethod';
export * from './ApiGen_Coordinate';
export * from './ApiGen_Document';
export * from './ApiGen_DocumentMetadataUpdate';
export * from './ApiGen_DocumentRelationship';
export * from './ApiGen_DocumentRelationType';
export * from './ApiGen_DocumentType';
export * from './ApiGen_EntityNote';
export * from './ApiGen_ExpropriationPayment';
export * from './ApiGen_ExpropriationPaymentItem';
export * from './ApiGen_File';
export * from './ApiGen_FinancialCode';
export * from './ApiGen_FinancialCodeTypes';
export * from './ApiGen_FormDocumentFile';
export * from './ApiGen_FormDocumentType';
export * from './ApiGen_Geometry';
export * from './ApiGen_H120Category';
export * from './ApiGen_Insurance';
export * from './ApiGen_InterestHolder';
export * from './ApiGen_InterestHolderProperty';
export * from './ApiGen_Lease';
export * from './ApiGen_LeaseTenant';
export * from './ApiGen_LeaseTerm';
export * from './ApiGen_Note';
export * from './ApiGen_NoteParent';
export * from './ApiGen_Organization';
export * from './ApiGen_OrganizationAddress';
export * from './ApiGen_OrganizationPerson';
export * from './ApiGen_Payment';
export * from './ApiGen_Person';
export * from './ApiGen_PersonAddress';
export * from './ApiGen_PersonOrganization';
export * from './ApiGen_Polygon';
export * from './ApiGen_Product';
export * from './ApiGen_Project';
export * from './ApiGen_ProjectProduct';
export * from './ApiGen_Property';
export * from './ApiGen_PropertyActivity';
export * from './ApiGen_PropertyActivityInvoice';
export * from './ApiGen_PropertyActivityInvolvedParty';
export * from './ApiGen_PropertyActivityProperty';
export * from './ApiGen_PropertyActivitySubtype';
export * from './ApiGen_PropertyAnomaly';
export * from './ApiGen_PropertyContact';
export * from './ApiGen_PropertyImprovement';
export * from './ApiGen_PropertyLease';
export * from './ApiGen_PropertyManagement';
export * from './ApiGen_PropertyManagementPurpose';
export * from './ApiGen_PropertyMinistryContact';
export * from './ApiGen_PropertyPurpose';
export * from './ApiGen_PropertyRoad';
export * from './ApiGen_PropertyTenure';
export * from './ApiGen_RegionUser';
export * from './ApiGen_ResearchFile';
export * from './ApiGen_ResearchFileProject';
export * from './ApiGen_ResearchFileProperty';
export * from './ApiGen_ResearchFilePurpose';
export * from './ApiGen_Role';
export * from './ApiGen_RoleClaim';
export * from './ApiGen_SecurityDeposit';
export * from './ApiGen_SecurityDepositReturn';
export * from './ApiGen_Take';
export * from './ApiGen_Type';
export * from './ApiGen_User';
export * from './ApiGen_UserRole';
