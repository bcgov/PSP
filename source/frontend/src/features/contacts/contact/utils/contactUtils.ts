import { CountryCodes } from 'constants/index';
import { AddressField } from 'features/contacts/interfaces';
import { validateYupSchema, yupToFormErrors } from 'formik';
import { IEditableOrganizationForm, IEditablePersonForm } from 'interfaces/editable-contact';
import isEmpty from 'lodash/isEmpty';

import {
  hasAddress,
  hasEmail,
  hasPhoneNumber,
  OrganizationValidationSchema,
  PersonValidationSchema,
} from '../create/validation';
import { sortAddresses } from '../detail/utils';
import { IContactAddress } from './../../../../interfaces/IContact';

export const onValidateOrganization = (
  values: IEditableOrganizationForm,
  otherCountryId?: string,
) => {
  const errors = {} as any;
  try {
    // combine yup schema validation with custom rules
    if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
      errors.needsContactMethod =
        'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
    }
    validateYupSchema(values, OrganizationValidationSchema, true, {
      otherCountry: otherCountryId,
    });
    return errors;
  } catch (err) {
    return { ...errors, ...yupToFormErrors(err) };
  }
};

export const onValidatePerson = (values: IEditablePersonForm, otherCountryId?: string) => {
  const errors = {} as any;
  try {
    // combine yup schema validation with custom rules
    if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
      errors.needsContactMethod =
        'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
    }
    validateYupSchema(values, PersonValidationSchema, true, { otherCountry: otherCountryId });

    return errors;
  } catch (err) {
    return { ...errors, ...yupToFormErrors(err) };
  }
};

export const toAddressFields = (addresses: IContactAddress[]) => {
  return addresses
    .sort(sortAddresses)
    .reduce((accumulator: AddressField[], value: IContactAddress) => {
      accumulator.push({
        label: value.addressType.description || '',
        streetAddress1: value.streetAddress1,
        streetAddress2: value.streetAddress2,
        streetAddress3: value.streetAddress3,
        municipalityAndProvince: [
          value.municipality !== undefined ? value.municipality : '',
          value.province?.provinceStateCode ?? '',
        ]
          .filter(x => !isEmpty(x))
          .join(' '),
        country:
          value.country?.countryCode === CountryCodes.Other
            ? value.countryOther ?? ''
            : value.country?.description,
        postal: value.postal,
      });
      return accumulator;
    }, []);
};
