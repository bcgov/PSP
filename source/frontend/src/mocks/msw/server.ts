import { setupServer } from 'msw/node';

import { handlers } from './handlers';

// This configures a request mocking server with the given request handlers (for jest unit tests).
const server = setupServer(...handlers);

export { server };
