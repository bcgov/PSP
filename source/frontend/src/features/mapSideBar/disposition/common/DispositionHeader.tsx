import React from 'react';
import { Col } from 'react-bootstrap';

import AuditSection from '@/components/common/HeaderField/AuditSection';
import { HeaderField } from '@/components/common/HeaderField/HeaderField';
import StatusField from '@/components/common/HeaderField/StatusField';
import { StyledFiller, StyledRow } from '@/components/common/HeaderField/styles';
import { Api_LastUpdatedBy } from '@/models/api/File';
import { ApiGen_Concepts_DispositionFile } from '@/models/api/generated/ApiGen_Concepts_DispositionFile';
import { exists } from '@/utils';

import HistoricalNumbersContainer from '../../shared/header/HistoricalNumberContainer';
import HistoricalNumberFieldView from '../../shared/header/HistoricalNumberSectionView';

export interface IDispositionHeaderProps {
  dispositionFile?: ApiGen_Concepts_DispositionFile;

  lastUpdatedBy: Api_LastUpdatedBy | null;
}

export const DispositionHeader: React.FunctionComponent<
  React.PropsWithChildren<IDispositionHeaderProps>
> = ({ dispositionFile, lastUpdatedBy }) => {
  const leftColumnWidth = '7';
  const leftColumnLabel = '3';

  const propertyIds = dispositionFile?.fileProperties?.map(fp => fp.propertyId) ?? [];

  return (
    <StyledRow className="no-gutters">
      <Col xs={leftColumnWidth}>
        <HeaderField label="File:" labelWidth={leftColumnLabel} contentWidth="9">
          D-{dispositionFile?.fileNumber}
        </HeaderField>
        <HistoricalNumbersContainer
          propertyIds={propertyIds}
          displayValuesOnly={false}
          View={HistoricalNumberFieldView}
        />
      </Col>
      <Col xs="5">
        <StyledFiller>
          <AuditSection lastUpdatedBy={lastUpdatedBy} baseAudit={dispositionFile} />
          {exists(dispositionFile?.fileStatusTypeCode) && (
            <StatusField statusCodeType={dispositionFile.fileStatusTypeCode} />
          )}
        </StyledFiller>
      </Col>
    </StyledRow>
  );
};

export default DispositionHeader;
