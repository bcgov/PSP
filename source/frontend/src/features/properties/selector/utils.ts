import { IMapProperty } from 'features/properties/selector/models';
import { compact } from 'lodash';
import { Api_PropertyFile } from 'models/api/PropertyFile';
import { formatApiAddress, pidFormatter } from 'utils';

import { Api_Geometry, Api_Property } from './../../../models/api/Property';

export enum NameSourceType {
  PID = 'PID',
  PIN = 'PIN',
  PLAN = 'Plan #',
  LOCATION = 'Location',
  NAME = 'Descriptive Name',
  NONE = 'n/a',
  ADDRESS = 'Address',
}

interface PropertyName {
  label: NameSourceType;
  value: string;
}

export const getPropertyName = (property: IMapProperty): PropertyName => {
  if (property.pid !== undefined && property.pid?.toString().length > 0 && property.pid !== '0') {
    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
  } else if (
    property.pin !== undefined &&
    property.pin?.toString()?.length > 0 &&
    property.pin !== '0'
  ) {
    return { label: NameSourceType.PIN, value: property.pin.toString() };
  } else if (property.planNumber !== undefined) {
    return { label: NameSourceType.PLAN, value: property.planNumber };
  } else if (property.latitude !== undefined && property.longitude !== undefined) {
    return {
      label: NameSourceType.LOCATION,
      value: compact([property.longitude?.toFixed(6), property.latitude?.toFixed(6)]).join(', '),
    };
  } else if (property.address !== undefined) {
    return {
      label: NameSourceType.ADDRESS,
      value: property.address,
    };
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getPrettyLatLng = (location?: Api_Geometry) =>
  compact([
    location?.coordinate?.x?.toFixed(6) ?? 0,
    location?.coordinate?.y?.toFixed(6) ?? 0,
  ]).join(', ');

export const getFilePropertyName = (
  fileProperty?: Api_PropertyFile,
  skipName: boolean = false,
): PropertyName => {
  if (fileProperty === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  if (
    fileProperty.propertyName !== undefined &&
    fileProperty.propertyName !== '' &&
    skipName === false
  ) {
    return { label: NameSourceType.NAME, value: fileProperty.propertyName };
  } else if (fileProperty.property !== undefined) {
    const property = fileProperty.property;
    return getApiPropertyName(property);
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getApiPropertyName = (property?: Api_Property): PropertyName => {
  if (property === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  let mapProperty: IMapProperty = {
    pin: property.pin?.toString(),
    pid: property.pid?.toString(),
    latitude: property.latitude,
    longitude: property.longitude,
    planNumber: property.planNumber,
    address: property.address !== undefined ? formatApiAddress(property.address) : undefined,
  };
  return getPropertyName(mapProperty);
};
