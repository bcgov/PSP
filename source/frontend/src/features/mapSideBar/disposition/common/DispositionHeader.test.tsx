import { mockDispositionFileResponse } from '@/mocks/dispositionFiles.mock';
import { server } from '@/mocks/msw/server';
import { getUserMock } from '@/mocks/user.mock';
import { ApiGen_Concepts_DispositionFile } from '@/models/api/generated/ApiGen_Concepts_DispositionFile';
import { prettyFormatUTCDate } from '@/utils/dateUtils';
import { render, RenderOptions } from '@/utils/test-utils';

import DispositionHeader, { IDispositionHeaderProps } from './DispositionHeader';
import { http, HttpResponse } from 'msw';

describe('DispositionHeader component', () => {
  // render component under test
  const setup = (props: IDispositionHeaderProps, renderOptions: RenderOptions = {}) => {
    const utils = render(
      <DispositionHeader
        dispositionFile={props.dispositionFile}
        lastUpdatedBy={props.lastUpdatedBy}
      />,
      {
        ...renderOptions,
      },
    );

    return { ...utils };
  };

  beforeEach(() => {
    server.use(
      http.get('/api/users/info/*', () => HttpResponse.json(getUserMock(), { status: 200 })),
    );
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('renders as expected when no data is provided', () => {
    const { asFragment } = setup({ lastUpdatedBy: null });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as expected when a disposition file is provided', async () => {
    const testDispositionFile = mockDispositionFileResponse();
    const { getByText } = setup({
      dispositionFile: testDispositionFile as unknown as ApiGen_Concepts_DispositionFile,
      lastUpdatedBy: {
        parentId: testDispositionFile.id || 0,
        appLastUpdateUserid: testDispositionFile.appLastUpdateUserid || '',
        appLastUpdateUserGuid: testDispositionFile.appLastUpdateUserGuid || '',
        appLastUpdateTimestamp: testDispositionFile.appLastUpdateTimestamp || '',
      },
    });

    expect(getByText(/FILE_NUMBER 3A8F46B/)).toBeVisible();
    expect(getByText(prettyFormatUTCDate(testDispositionFile.appCreateTimestamp))).toBeVisible();
    expect(
      getByText(prettyFormatUTCDate(testDispositionFile.appLastUpdateTimestamp)),
    ).toBeVisible();
  });

  it('renders the file number and name concatenated', async () => {
    const testDispositionFile = mockDispositionFileResponse();
    const { getByText } = setup({ dispositionFile: testDispositionFile, lastUpdatedBy: null });

    expect(getByText('File:')).toBeVisible();
    expect(getByText(/FILE_NUMBER 3A8F46B/)).toBeVisible();
  });

  it('renders the last-update-time when provided', async () => {
    const testDate = new Date().toISOString();
    const testDispositionFile = mockDispositionFileResponse();
    const { getByText } = setup({
      dispositionFile: testDispositionFile,
      lastUpdatedBy: {
        parentId: testDispositionFile.id || 0,
        appLastUpdateUserid: 'Test User Id',
        appLastUpdateUserGuid: 'TEST GUID',
        appLastUpdateTimestamp: testDate,
      },
    });

    expect(getByText(/FILE_NUMBER 3A8F46B/)).toBeVisible();
    expect(getByText(prettyFormatUTCDate(testDispositionFile.appCreateTimestamp))).toBeVisible();
    expect(getByText(prettyFormatUTCDate(testDate))).toBeVisible();
  });

  it('renders the file status when provided', async () => {
    const testDispositionFile: ApiGen_Concepts_DispositionFile = {
      ...mockDispositionFileResponse(),
      fileStatusTypeCode: {
        id: 'TEST',
        description: 'mock file status',
        displayOrder: null,
        isDisabled: false,
      },
    };
    const { getByText } = setup({ dispositionFile: testDispositionFile, lastUpdatedBy: null });

    expect(getByText('Status:')).toBeVisible();
    expect(getByText(/mock file status/i)).toBeVisible();
  });
});
