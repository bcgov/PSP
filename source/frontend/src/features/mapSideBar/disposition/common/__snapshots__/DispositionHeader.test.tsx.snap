// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DispositionHeader component > renders as expected when no data is provided 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c2 {
  font-size: 0.87em;
  line-height: 1.9;
}

.c0 {
  margin-top: 0.5rem;
  margin-bottom: 1.5rem;
  border-bottom-style: solid;
  border-bottom-color: grey;
  border-bottom-width: 0.1rem;
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    class="c0 no-gutters row"
  >
    <div
      class="col-7"
    >
      <div
        class="row"
      >
        <div
          class="pr-0 text-left col-3"
        >
          <strong>
            <label>
              File:
            </label>
          </strong>
        </div>
        <div
          class="pl-1 text-left col-9"
        >
          D-
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="pr-0 text-left col-3"
        >
          <strong>
            <label>
              Historical File #:
            </label>
          </strong>
        </div>
        <div
          class="pl-1 text-left col-9"
        >
          <div />
        </div>
      </div>
    </div>
    <div
      class="col-5"
    >
      <div
        class="c1"
      >
        <div
          class="no-gutters row"
        >
          <div
            class="text-right col"
          >
            <span
              class="c2"
            >
              <strong>
                Created: 
              </strong>
               by 
              <span
                class="tooltip-icon"
                data-testid="tooltip-icon-userNameTooltip"
                id="userNameTooltip"
              >
                <strong>
                  USER
                </strong>
              </span>
            </span>
          </div>
        </div>
        <div
          class="no-gutters row"
        >
          <div
            class="text-right col"
          >
            <span
              class="c2"
            >
              <strong>
                Updated: 
              </strong>
               by 
              <span
                class="tooltip-icon"
                data-testid="tooltip-icon-userNameTooltip"
                id="userNameTooltip"
              >
                <strong>
                  USER
                </strong>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
